server/neptune/gateway/event/pull_request_review_handler.go

```diff
@@ -29,25 +34,41 @@ type fetcher interface {
 	ListFailedPolicyCheckRunNames(ctx context.Context, installationToken int64, repo models.Repo, ref string) ([]string, error)
 }
 
-type PullRequestReviewWorkerProxy struct {
-	Scheduler       scheduler
-	SnsWriter       Writer
-	Logger          logging.Logger
-	CheckRunFetcher fetcher
+type prApprovalSignaler interface {
+	SignalWorkflow(ctx context.Context, workflowID string, runID string, signalName string, arg interface{}) error
 }
 
-func (p *PullRequestReviewWorkerProxy) Handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
-	return p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
-		return p.handle(ctx, event, request)
-	})
+type PullRequestReviewWorkerProxy struct {
+	Scheduler          scheduler
+	SnsWriter          Writer
+	Logger             logging.Logger
+	CheckRunFetcher    fetcher
+	Allocator          feature.Allocator
+	PRApprovalSignaler prApprovalSignaler
+	Scope              tally.Scope
 }
 
-func (p *PullRequestReviewWorkerProxy) handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
+func (p *PullRequestReviewWorkerProxy) Handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
 	// Ignore non-approval events
 	if event.State != Approved {
 		return nil
 	}
+	err := p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		if legacyErr := p.handleLegacyMode(ctx, request, event); legacyErr != nil {
+			p.Logger.ErrorContext(ctx, legacyErr.Error())
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
i think instead of this I'd prefer something like this:
https://pkg.go.dev/github.com/hashicorp/go-multierror

I don't think these should be dependent on one succeeding, they're independent events.

Reply:
Actually, I realized this err isn't really needed at all because the asynchronous scheduler delegates logging errors and always returns `nil`: https://github.com/lyft/atlantis/blob/main/server/neptune/sync/scheduler.go#L36-L47

Reply:
Probably not a great idea to assume the implementation acts one way with an interface tbh. Can we just use something like a multi-err to be more resilient to an unexpected change?

Reply:
Sure thing, but I assumed it was intentional set up that way to wrap async around sync scheduler. I'll skip duplicating error logging though.